docs/instructions.md

```diff
@@ -62,16 +79,94 @@ env = android_env.load(
 )
 ```
 
-## Example scripts
+## Example RL agent scripts
+
+The `examples` directory contains a few simple example agent setups, such as:
+
+*   [`run_random_agent.py`](https://github.com/deepmind/android_env/blob/main/examples/run_random_agent.py):
+    Runs a simple loop performing randomly selected actions in the environment.
+*   [`run_acme_agent.py`](https://github.com/deepmind/android_env/blob/main/examples/run_acme_agent.py):
+    Runs a training loop with an [Acme](https://deepmind.com/research/publications/Acme)
+    DQN agent, implemented in the popular DeepMind RL framework. This will
+    require to install the [`acme`](https://github.com/deepmind/acme)
+    dependency.
+*   [`run_human_agent.py`](https://github.com/deepmind/android_env/blob/main/examples/run_human_agent.py):
+    Creates a [`pygame`](https://www.pygame.org) instance that lets a human user
+    interact with the environment and observe environment mechanics, such as
+    rewards or task extras. You will need to install the [PyGame] dependency.
+
+For instance, here is how you can run `run_random_agent.py` in a folder where
+you have your APK file, such as
+the [Apple Flinger](https://github.com/deepmind/android_env/blob/main/docs/example_tasks.md#apple-flinger)
+game.
+(You can find the TAR file with the APK and `.textproto` files
+[here](https://storage.googleapis.com/android_env-tasks/apple_flinger.tar.gz)
+and on the
+[example tasks](https://github.com/deepmind/android_env/blob/main/docs/example_tasks.md)
+page.)
+
+```shell
+python3 run_random_agent.py \
+--avd_name='my_avd',
+--android_avd_home=/Users/username/.android/avd \
+--android_sdk_root=/Users/username/Library/Android/sdk \
+--emulator_path=/Users/username/Library/Android/sdk/emulator/emulator \
+--adb_path=/Users/username/Library/Android/sdk/platform-tools/adb \
+--num_steps=1000 \
+--task_path=/Users/username/<PATH-TO-APPLE-FLINGER-FILES>/apple_flinger_M_1_1.textproto
+```
+
+## Example with code: a random agent playing Droidfish (chess)
```

Comment:
I'm not sure if it's better to inline the code here in Markdown vs just pointing the user to the example script. If they want to run that anyway, they'll need to interact with the command-line, and if they just want to see the code, they can just open the script.
The script may also deviate from this example in the future, so I prefer to just keep the `.py` file.

Reply:
The code in the example scripts (random play, human play, and Acme) seem intuitive IMO. 

The non-intuitive parts in my experience have been getting the configuration right for the Android stuff's locations (maybe because we're interacting with the AVD outside of Android Studio, it isn't easy). Therefore, I think we should hold the users' hands a little and _minimize the number of clicks_ to get their first "Hello, World/Frozen-Lake" RL experiment running, which may be important for the env adoption. I'm thinking about the users who wouldn't be spending as much time as us researching an awesome new RL environment.

_Initial investment:_ The user, who we can assume is not likely to be a regular mobile app dev, up to this point has had to go through several steps just to have Studio and AVD set up (downloading the software and then some extras can be time-consuming). The AVD emulator will hopefully run on most machines with Intel/AMD or similar (there're cases, when it will fail, so they'd have to resort to running on a physical device).

_More UX friction:_ Then, the user would have to remember the various locations of the following to get a "Hello, World" example up and running (and that's before they pick a TAR file to unzip with an app and some example tasks):

1. the emulator AVD location
2. the SDK location (the default one varies by OS)
3. the emulator + ADB paths, which are nested in the SDK folder
4. the path to `.textproto` Task (and don't forget to download and unzip the TAR file)
5. ❕ the unzipped APK location (i.e. where to run `some_script.py`): this was not easy to figure out initially, even though I had several APKs preinstalled already - users may run a script only to find out that once the AVD is ready (after booting up), the script crashes, because the APK file is not detected (even though the app has been pre-installed).

The amount of friction is probably not ideal if someone wants to quickly "do RL on Frozen-Lake", but this is for research #science.

_Conclusion:_ Based on this experience, I'd recommend having some sort of example scripts clearly spelled out without the user having to click and explore the repo themselves. This also means we'd have to maintain them, if something changes, but they are quite short.

The current proposal is similar to what Retro Gym managed to achieve (https://retro.readthedocs.io/en/latest/getting_started.html): 

- One example for the command line interface, e.g.: 

`python3 -m retro.examples.random_agent --game some_game_ROM`
 
- And one example with some example code, such as:

```python
import retro

def main():
    env = retro.make(game='some_game_ROM')
    obs = env.reset()
    while True:
        obs, rew, done, info = env.step(env.action_space.sample())
        env.render()
        if done:
            obs = env.reset()
    env.close()


if __name__ == "__main__":
    main()
```

Anyway, these are my 2 cents 🙂 LMKWYT and I'll do what you think is right. 

Thanks again for the awesome project!

Reply:
Hello! We still prefer not to duplicate code inside the documentation. Please remove the section "Example with code: a random agent playing Droidfish (chess)". Thanks!

Reply:
Definitely, will do! Thanks again 🙂